metamngmt/msgtypes.go

```diff
@@ -59,10 +60,13 @@ type AddDerived struct {
 	Nonce []byte `json:"nonce"`
 
 	Tangles refs.Tangles `json:"tangles"`
+	QueryLang string `json:"querylang"`
+	Query string `json:"query"`
 }
 
 // NewAddDerivedMessage just initializes type and the passed fields.
-// Callers need to set the right tangle point themselves afterwards.
+// Callers need to set the right tangle point themselves afterwards. If the message has metadata that needs to be added,
+// function AddDerived.InsertMetadata should be used.
 func NewAddDerivedMessage(meta, sub refs.FeedRef, purpose string, nonce []byte) AddDerived {
```

Comment:
```suggestion
func NewAddDerivedMessage(meta, sub refs.FeedRef, purpose string, nonce []byte, metadata ...map[string]string) (AddDerived, error) {

  if len(metadata) > 0 {

    if len(metadata) > 1 {
      return errors.New("only supports one metadata value")
    }

    err:=insertMetadata(metadata[0])
    // if err...

  }
...
```

This way we don't have to add `InsertMetadata()`.